axlearn/common/trainer_config_modifier.py

```diff
@@ -0,0 +1,151 @@
+"""Defines trainer config modifiers, which will be used in model definitions."""
+
+from typing import Dict, List, Optional, Union
+
+from axlearn.common import config
+from axlearn.common.base_layer import RematSpec
+from axlearn.common.config import REQUIRED, ConfigModifier, ConfigOr, Required, config_class
+from axlearn.common.gradient_accumulation import with_minibatch_steps
+from axlearn.common.metrics import MetricAccumulator
+from axlearn.common.trainer import SpmdTrainer
+from axlearn.common.utils import HybridMeshShape, MeshShape
+
+
+class GradientAccumulation(ConfigModifier):
+    """Accumulate gradients for grad_acc_steps steps."""
+
+    @config_class
+    class Config(ConfigModifier.Config):
+        grad_acc_steps: Optional[int] = None
+        metric_accumulator: Required[MetricAccumulator.Config] = MetricAccumulator.default_config()
+
+    def __init__(self, cfg: Config):
+        super().__init__(cfg)
+        cfg = self.config
+        self._grad_acc_steps = cfg.grad_acc_steps
+        self._metric_accumulator = cfg.metric_accumulator
+
+    def __call__(self, cfg: SpmdTrainer.Config) -> SpmdTrainer.Config:
+        """Overwrite the forward_fn_transformation to accumulate gradients for grad_acc_steps steps.
+
+        Note this would not affect the global batch size or the logical training steps.
+        The optimization step is applied each time after grad_acc_steps steps of
+        forward and backward passes on mini-batches.
+
+        global_bs=mini_bs*grad_acc_steps
+        train_steps=mini_steps/grad_acc_steps
+
+        Args:
+            cfg (SpmdTrainer.Config): the trainer config to be modified.
+
+        Returns:
+            SpmdTrainer.Config: the modified trainer config.
+        """
+        if self._grad_acc_steps is None:
+            return cfg
+
+        cfg.learner.forward_fn_transformation = config.config_for_function(
+            with_minibatch_steps
+        ).set(
+            steps=self._grad_acc_steps,
+            metric_accumulator=self._metric_accumulator,
+        )
+        return cfg
+
+
+class RematPolicies(ConfigModifier):
```

Comment:
A suggestion on naming:
```suggestion
class RematSpecModifier(ConfigModifier):
```

It may be more consistent/discoverable to name them as `XModifier`.